recipes/cpp-peglib/1.x.x/conanfile.py

```diff
@@ -0,0 +1,67 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class CpppeglibConan(ConanFile):
+    name = "cpp-peglib"
+    description = "A single file C++11 header-only PEG (Parsing Expression Grammars) library."
+    license = "MIT"
+    topics = ("conan", "cpp-peglib", "peg", "parser", "header-only")
+    homepage = "https://github.com/yhirose/cpp-peglib"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15.7",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10"
+        }
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
+
+        def lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("{} {} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(self.name, self.version))
+        elif lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
+            raise ConanInvalidConfiguration("{} {} requires C++17, which your compiler does not support.".format(self.name, self.version))
+
+        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) == "7" and \
```

Comment:
_compilers_minimum_version declares minimal version 6 for clang. Should we set stdcpp for clang 6 too?

Reply:
No, cpp-peglib works fine with clang 6 and libstdc++. I guess it's a bug in clang 7.